src/content/en/2021/media.md

```diff
@@ -556,12 +556,10 @@ The `<picture>` element serves a couple of use cases:
 
 Art direction appears a bit more popular than format-switching, but both features appear underutilized when you consider their potential utility. As we've seen, very few pages are tailoring images' aspect ratios to fit mobile screens, and many more pages could deliver their imagery more efficiently using next-generation formats. These are exactly the problems that `<picture>` was invented to solve, and perhaps more than 5.9% of pages could be addressing them, using it.
 
-It's possible that format-switching with `<source type>` is only used on 2-3% of pages because format-switching can also be accomplished using [server-side content negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation). Unfortunately server-side adaptation mechanisms are hard to detect in the crawled data, and we have not analyzed them here.
+It's possible that format-switching with `<source type>` is only used on 2-3% of pages because format-switching can also be accomplished using [server-side content negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation). Unfortunately, server-side adaptation mechanisms are hard to detect in the crawled data, and we have not analyzed them here.
 
 Notably, `<source type>` and `<source media>` are not mutually exclusive, and, put together, the usage percentages here do not add up to 100%. This suggests that at least 15% of `<picture>` elements do not leverage either of these attributes, making those `<picture>`s functionally equivalent to a `<span>`.
 
-{# TODO consdier investigating a few examples to see if can understand why #}
-
```

Comment:
@eeeps I had a look at this.

Here's an example (from [this page](https://tokyomart.com.au/)):

```html
<picture class="loaded" style="display: none;">
    <source
        data-srcset=" //cdn.shopify.com/s/files/1/0461/8154/8197/products/Screen_Shot_2021-11-24_at_9.55.34_am-removebg-preview_180x.png?v=1637708338 180w 170h, //cdn.shopify.com/s/files/1/0461/8154/8197/products/Screen_Shot_2021-11-24_at_9.55.34_am-removebg-preview_360x.png?v=1637708338 360w 340h, "
        sizes="260px"
        srcset=" //cdn.shopify.com/s/files/1/0461/8154/8197/products/Screen_Shot_2021-11-24_at_9.55.34_am-removebg-preview_180x.png?v=1637708338 180w 170h, //cdn.shopify.com/s/files/1/0461/8154/8197/products/Screen_Shot_2021-11-24_at_9.55.34_am-removebg-preview_360x.png?v=1637708338 360w 340h, ">
    <img alt="" class="lazyautosizes lazyloaded" data-sizes="auto" data-parent-fit="contain" sizes="260px">
</picture>
```

So we've a few things here:
1. The img src is empty (nice fallback - not!)
2. Looks like it's using lazy sizes
3. Looks like it's using `srcset` to provide the media ability


I ran this query:

```sql
SELECT
  _TABLE_SUFFIX AS client,
  COUNT(0) AS total_such_images,
  ROUND(COUNTIF(JSON_EXTRACT_SCALAR(image, '$.url') LIKE '%shopify%') / COUNT(0) * 100, 2) AS pct_shopifyURL,
  ROUND(COUNTIF(JSON_EXTRACT_SCALAR(image, '$.hasSrc') = 'true') / COUNT(0) * 100, 2) AS pct_hasSrc,
  ROUND(COUNTIF(JSON_EXTRACT_SCALAR(image, '$.hasSrcset') = 'true') / COUNT(0) * 100, 2) AS pct_hasSrcset,
  ROUND(COUNTIF(JSON_EXTRACT_SCALAR(image, '$.hasSizes') = 'true') / COUNT(0) * 100, 2) AS pct_hasSizes,
  ROUND(COUNTIF(JSON_EXTRACT_SCALAR(image, '$.url') NOT LIKE '%shopify%' AND JSON_EXTRACT_SCALAR(image, '$.hasSrc') = 'false' AND JSON_EXTRACT_SCALAR(image, '$.hasSrcset') = 'false' AND JSON_EXTRACT_SCALAR(image, '$.hasSizes') = 'false') / COUNT(0) * 100, 2) AS pct_noneOfThese
FROM
  `httparchive.pages.2021_07_01_*`,
  UNNEST(JSON_EXTRACT_ARRAY(JSON_EXTRACT_SCALAR(payload, '$._responsive_images'), '$.responsive-images')) AS image
WHERE
  JSON_EXTRACT_SCALAR(image, '$.isInPicture') = 'true' AND
  JSON_EXTRACT_SCALAR(image, '$.pictureMediaSwitching') = 'false' AND
  JSON_EXTRACT_SCALAR(image, '$.pictureTypeSwitching') = 'false'
GROUP BY
  client

```

And it comes up with this:

client | total_such_images | pct_shopifyURL | pct_hasSrc | pct_hasSrcset | pct_hasSizes | pct_noneOfThese
-- | -- | -- | -- | -- | -- | --
desktop | 1276092 | 22.65 | 69.32 | 39.32 | 29.2 | 7.42
mobile | 1320529 | 19.43 | 71.87 | 35.02 | 25.33 | 8.6

Any thoughts on this? Or anything more you want to add to this paragraph now we know this?

Reply:
@tunetheweb Interesting!

Re: no src -- this is the reccomended implementation pattern for lazysizes. Including a `src` would incur a double-load for most people (of ① the `src` and ② the resource ultimately chosen out of the `srcset` [which is initially not present until JS runs and it's copied over from the `data-srcset`]). So lazysizes made the (reasonable, given the constraints) choice to suggest alt text as a fallback, for chases when it doesn't run. As you can see here: many sites that use it don't provide alt text. But I digress.

I think previous Media chapter author @colinbendell is actually in charge of what Shopify is doing here, so perhaps he can chime in, but my best guess is that this is the output of some sort of a generic "image" template. That template has been built to be flexible enough to do art direction, but often isn't called to do so. But when it isn't, (e.g., here) the `<source>` contains a copy of what's already available in the `<img>`, and the `<picture>` effectively does nothing, resource-selection wise. But we can't remove the function-less `<picture>` container, as styling has possibly been built on top of it.

I could venture this guess in the chapter prose, but it would be just that: a guess.

It's fascinating to me that Shopify, alone is responsible for 1 in 5 of these images, though. Do you think that's worth including?

Reply:
Let's leave as is. Your original conclusion ("making those `<picture>`s functionally equivalent to a `<span>`.") is still accurate.